src/main/java/frc/robot/poseestimator/helpers/ObservationCountHelper.java

```diff
@@ -0,0 +1,38 @@
+package frc.robot.poseestimator.helpers;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+
+public class ObservationCountHelper<T> {
```

Comment:
Observation count? why? is it only used to count observations?
Not only that it can be used on other things that are not observations, it doesn't even count them!
fix the name of the class please🙏

Reply:
```suggestion
public class ObservationAccumulator<T> {
```

?